doc/nrf/libraries/bluetooth_services/mesh/sensor.rst

```diff
@@ -333,6 +353,50 @@ Each entry in the list has a type and two access callbacks, and the list should
 The :c:member:`bt_mesh_sensor_setting.get` callback is mandatory, while the :c:member:`bt_mesh_sensor_setting.set` is optional, allowing for read-only entries.
 The value of the settings may change at runtime, even outside the ``set`` callback.
 New values may be rejected by returning a negative error code from the ``set`` callback.
+See a small sample of adding a setting to a sensor below:
+
+.. code-block:: c
+
+   static void motion_threshold_get(struct bt_mesh_sensor_srv *srv,
+                                    struct bt_mesh_sensor *sensor,
+                                    const struct bt_mesh_sensor_setting *setting,
+                                    struct bt_mesh_msg_ctx *ctx,
+                                    struct sensor_value *rsp)
+   {
+        /** Get the current threshold in an application defined way and
+         *  store it in rsp.
+         */
+        get_threshold(rsp);
+   }
+
+   static int motion_threshold_set(struct bt_mesh_sensor_srv *srv,
+                                   struct bt_mesh_sensor *sensor,
+                                   const struct bt_mesh_sensor_setting *setting,
+                                   struct bt_mesh_msg_ctx *ctx,
+                                   const struct sensor_value *value)
+   {
+        /** Store incoming threshold in application-defined way.
+         *  Return error code to reject set.
+         */
+        return set_threshold(value);
+   }
+
+   static struct bt_mesh_sensor_setting settings[] = {
```

Comment:
Can this be const? And then remove the typecast on line 396.

Reply:
This was a lazy code-grap from the sensor_server sample application on my part, the typecast is not needed even if this is not const, the array can be passed bare and will degrade to a (const) pointer just fine. Will update with more readable code.